File: backend/src/main/java/project/lms_rikkei_edu/modules/csvimport/service/CsvImportService.java
package project.lms_rikkei_edu.modules.csvimport.service;

import org.springframework.web.multipart.MultipartFile;
import project.lms_rikkei_edu.modules.csvimport.dto.response.CsvImportConfirmResponse;
import project.lms_rikkei_edu.modules.csvimport.dto.response.CsvImportPreviewResponse;

import java.util.List;
import java.util.UUID;

public interface CsvImportService {

    CsvImportPreviewResponse preview(MultipartFile file, String defaultRole, UUID courseId, List<UUID> groupIds);

    CsvImportConfirmResponse confirm(String token, UUID adminId, UUID courseId, List<UUID> groupIds);
}

File: frontend/src/services/csv-import-service.ts
import { httpClient } from '../lib';
import type { CsvImportConfirmResponse, CsvImportPreviewResponse } from '../types';

export async function previewCsvImport(file: File, defaultRole: string, courseId?: string, groupIds?: string[]) {
  const formData = new FormData();
  formData.append('file', file);
  formData.append('defaultRole', defaultRole);
  if (courseId) formData.append('courseId', courseId);
  if (groupIds && groupIds.length > 0) {
    groupIds.forEach(id => formData.append('groupIds', id));
  }
  const response = await httpClient.post<CsvImportPreviewResponse>(
    '/admin/users/import-csv/preview',
    formData,
    { headers: { 'Content-Type': 'multipart/form-data' } },
  );
  return response.data;
}

export async function confirmCsvImport(token: string, courseId?: string, groupIds?: string[]) {
  const response = await httpClient.post<CsvImportConfirmResponse>(
    '/admin/users/import-csv/confirm',
    { token, courseId, groupIds },
  );
  return response.data;
}
